pkg/controller/build/imagebuilder/base.go

```diff
@@ -278,35 +278,35 @@ type kubeObject interface {
 // Computes the MachineOSBuild status given the build status as well as the
 // conditions. Also fetches the final image pullspec from the digestfile
 // ConfigMap.
-func (b *baseImageBuilder) getMachineOSBuildStatus(ctx context.Context, obj kubeObject, buildStatus mcfgv1alpha1.BuildProgress, conditions []metav1.Condition) (mcfgv1alpha1.MachineOSBuildStatus, error) {
+func (b *baseImageBuilder) getMachineOSBuildStatus(ctx context.Context, obj kubeObject, buildStatus mcfgv1.BuildProgress, conditions []metav1.Condition) (mcfgv1.MachineOSBuildStatus, error) {
 	now := metav1.Now()
 
-	out := mcfgv1alpha1.MachineOSBuildStatus{}
+	out := mcfgv1.MachineOSBuildStatus{}
 
 	out.BuildStart = &now
 
-	if buildStatus == mcfgv1alpha1.MachineOSBuildSucceeded || buildStatus == mcfgv1alpha1.MachineOSBuildFailed || buildStatus == mcfgv1alpha1.MachineOSBuildInterrupted {
+	if buildStatus == mcfgv1.MachineOSBuildSucceeded || buildStatus == mcfgv1.MachineOSBuildFailed || buildStatus == mcfgv1.MachineOSBuildInterrupted {
 		out.BuildEnd = &now
 	}
 
-	if buildStatus == mcfgv1alpha1.MachineOSBuildSucceeded {
+	if buildStatus == mcfgv1.MachineOSBuildSucceeded {
 		pullspec, err := b.getFinalImagePullspec(ctx)
 		if err != nil {
 			return out, err
 		}
 
-		out.FinalImagePushspec = pullspec
+		out.DigestedImagePushSpec = mcfgv1.ImageDigestFormat(pullspec)
 	}
 
 	out.Conditions = conditions
-	out.BuilderReference = &mcfgv1alpha1.MachineOSBuilderReference{
-		ImageBuilderType: mcfgv1alpha1.PodBuilder,
+	out.Builder = &mcfgv1.MachineOSBuilderReference{
+		ImageBuilderType: mcfgv1.JobBuilder,
 		// TODO: Should we clear this whenever the build is complete?
-		PodImageBuilder: &mcfgv1alpha1.ObjectReference{
+		Job: &mcfgv1.ObjectReference{
 			Name:      obj.GetName(),
-			Group:     obj.GroupVersionKind().Group,
+			Group:     "tbd-group",
 			Namespace: obj.GetNamespace(),
-			Resource:  obj.GetResourceVersion(),
+			Resource:  "tbd-resource",
```

Comment:
The values being applied fail API validation. The API requires that all these fields should validate against `format.dns1123Subdomain()`. In my testing Group seemed to be often assigned an empty value. Resource Version, which is normally a number, being assigned to Resource also seems like a mistake. Open to suggestions here!